migrations/1_initialize_schema.up.sql

```diff
@@ -463,18 +463,31 @@ CREATE TABLE item_attribute
   CHECK ((item_id IS NOT NULL AND launch_id IS NULL) OR (item_id IS NULL AND launch_id IS NOT NULL))
 );
 
-CREATE TABLE log
-(
-  id                   BIGSERIAL
-    CONSTRAINT log_pk PRIMARY KEY,
-  log_time             TIMESTAMP                                               NOT NULL,
-  log_message          TEXT                                                    NOT NULL,
-  item_id              BIGINT REFERENCES test_item (item_id) ON DELETE CASCADE NOT NULL,
-  last_modified        TIMESTAMP                                               NOT NULL,
-  log_level            INTEGER                                                 NOT NULL,
-  attachment           TEXT,
-  attachment_thumbnail TEXT,
-  content_type         TEXT
+CREATE TABLE attachment (
+  id             BIGSERIAL CONSTRAINT attachment_pk PRIMARY KEY,
+  file_id        TEXT NOT NULL,
+  thumbnail_id   TEXT,
+  content_type   TEXT,
+  project_id     BIGINT,
```

Comment:
why are they not foreign keys ?

Reply:
We need to remove all attachments after project/launch/item removing, attachments removing triggers after committing project/launch/item removing, so if we make it a foreign key attachment's record will be removed/foreign key will be set to NULL/removing of the project will be suppressed by constraint violation on the DB-level. Process of putting all attachments' records in the rabbit queue by stream is pretty time-consuming, so response after project/launch removing will be delayed. But if commit transaction after the project/launch/item removing and proceed attachments removing in the Transactional Event Listener, we can send a response to the client and then put all related attachments (by project/launch/item id) in the queue and remove them.